src/gwt/src/org/rstudio/core/client/VirtualConsole.java

```diff
@@ -580,6 +601,41 @@ public void submit(String data, String clazz, boolean forceNewRange, boolean ari
                // we don't support. Tricky part is we might get codes split across
                // submit calls.
 
+               // match hyperlink, either start or end (if [url] is empty
+               // <ESC> ] 8 ; [params] ; [url] \7
+               if (ansi_ == null)
+                  ansi_ = new AnsiCode();
+                     
+               Match hyperlinkMatch = AnsiCode.HYPERLINK_PATTERN.match(data.substring(pos), 0);
+               
+               if (hyperlinkMatch != null)
+               {
+                  String url = hyperlinkMatch.getGroup(2);
+                  // toggle hyperlink, and artifically add or remove styles: underline, magenta and italic
+                  if (!StringUtil.equals(url, ""))
+                  {
+                     ansiCodeStyles_ = ansi_.processCode("\033[4m"); // underline
+                     ansiCodeStyles_ = ansi_.processCode("\033[3m"); // italic
```

Comment:
IMHO we don't need italics for hyperlinks by default; just underlining + giving a separate color should be fine.

Reply:
Thanks. I removed it.